instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/coroutines/TestUtils.kt

```diff
@@ -68,10 +71,42 @@ suspend fun MapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes: List<Na
     }
 }
 
+suspend fun MapboxNavigation.setNavigationRoutesAndAwaitError(
+    routes: List<NavigationRoute>,
+    legIndex: Int
+) = withTimeout(MAX_TIME_TO_UPDATE_ROUTE) {
+    suspendCoroutine<Unit?> { continuation ->
```

Comment:
You should use `suspendCancellableCoroutine` even if you don't use `invokeOnCancellation`, otherwise `withTimeout` won't work. 
```suggestion
    suspendCancellableCoroutine<Unit?> { continuation ->
```